modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/app/ApplicationConfigOps.scala

```diff
@@ -34,6 +39,22 @@ object ConfigReaders {
   implicit val portReader: ConfigReader[Port] =
     ConfigReader[Int].emap(i => Port.fromInt(i).toRight(CannotConvert(i.toString, "Port", "Parse resulted in None")))
 
+  implicit val amountReader: ConfigReader[Amount] =
+    ConfigReader[Long].emap { i =>
+      RefType.applyRef[NonNegLong](i) match {
+        case Left(_) => CannotConvert(i.toString, "Amount", "Parse resulted in None").asLeft[Amount]
+        case Right(value) => Amount(value).asRight[CannotConvert]
+      }
+    }
```

Comment:
afaik `ConfigReader[Amount]` is available in the tessellation 🤔 you can reuse it